### PR TITLE
[FEATURE] Add support for friend link shortcode

### LINKED ISSUE #30: A possible friend links layout
I also copy from other guys, but I forget to save the author's link. If there's infringement, please tell me, since I can't contact with the author.
[My friend links webpage](https://www.summerfire.cn/friend/)

### PR DESCRIPTION
**Pull request by HEIGE-PCloud**
- Add friend link shortcode template
- Add friend link shortcode stylesheet
- Add documentation for friend link shortcode
- Add test page for friend link shortcode

Many thanks to this [page](https://blog.233so.com/2020/04/friend-link-shortcodes-for-hugo-loveit/).

### CODE REVIEW

**Comment by HEIGE-PCloud** [comment]
Check this [page](https://6bed22bb.hugodoit.pages.dev/friend-link-tests/) for preview.

**Comment by HEIGE-PCloud** [comment]
The mobile view needs some adjustment.

image

**Review by SpongeBab — DISMISSED** [review]

File: layouts/shortcodes/friend.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{{ if .IsNamedParams }}
+<a target="_blank" href={{ .Get "url" }} title={{ .Get "name" }} class="friend-link">
+    <div class="friend-link-div">
+        <div class="friend-link-avatar">
+            <img class="friend-avatar" src={{ .Get "avatar" }} />
```
Comment (HEIGE-PCloud) [inline layouts/shortcodes/friend.html:5]: Need to add support for lazyload. No bare image tag.

File: layouts/shortcodes/friend.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            <i class="friend-name">{{ .Get "name" }}</i>
+            <p class="friend-bio">{{ .Get "bio" }}</p>
+        </div>
+    </div>
+</a>
+{{- else -}}
+<a target="_blank" href={{ .Get 1 }} title={{ .Get 0 }} class="friend-link">
+    <div class="friend-link-div">
+        <div class="friend-link-avatar">
+            <img class="friend-avatar" src={{ .Get 2 }} />
```
Comment (HEIGE-PCloud) [inline layouts/shortcodes/friend.html:18]: Same issue here.

File: layouts/shortcodes/friend.html (hunk 1 above)
Comment (HEIGE-PCloud) [inline layouts/shortcodes/friend.html]: Need to add support for lazyload. No bare image tag.

File: layouts/shortcodes/friend.html (hunk 2 above)
Comment (HEIGE-PCloud) [inline layouts/shortcodes/friend.html]: Same issue here.

**Comment by HEIGE-PCloud** [comment]
@SpongeBab Review is not something that I click `review request`, you click `approved` then everything is fine and call it a day. Identifying bugs and issues at the pull request stage is much better than merging bugs into production and then open a new issue to fix it. I check my code today and very luckily fix the issue before clicking `Rebase and merge`, but I may fail to do that in future pull requests. I hope the reviews and approves can be more useful and have some real feedback. Many thanks.

**Comment by SpongeBab** [comment]
For some reason,I can not use npm. I use huho server and everything is OK.I think you should test and build the website by hugo because people use hugo to generate static website.
I’m sorry for I can’t give you more help.I haven’t used javascript, shortcode and HTML is out...... My tutor urged me to do a project and write a paper for him. Forgive me for not having time to learn something new used to develop a website. There used to be a lot of things that didn't learn. now it often feels that time is not enough.

时不利兮骓不逝。

Regard!



发自我的iPhone

**Pull request merged**